tests/src/vm_host_env/tx.rs

```diff
@@ -171,6 +180,22 @@ impl TestTxEnv {
             .write(&storage_key, public_key.try_to_vec().unwrap())
             .unwrap();
     }
+
+    /// Apply the tx changes to the write log and return
+    /// the set of verifiers.
+    pub fn execute_tx(&mut self) -> Result<(), Error> {
+        let empty_data = vec![];
+        wasm::run::tx(
+            &self.storage,
+            &mut self.write_log,
+            &mut self.gas_meter,
+            &self.tx.code,
+            self.tx.data.as_ref().unwrap_or(&empty_data),
+            &mut self.vp_wasm_cache,
+            &mut self.tx_wasm_cache,
+        )
+        .and(Ok(()))
+    }
```

Comment:
🎊 worth it's own PR back to `main`!

Reply:
https://github.com/anoma/namada/pull/738